### ISSUE TITLE
Can't create 'last used session' symlink

### ISSUE THREAD

**Issue by kadfak** [mhinz/vim-startify#243]
Hi,

I'm having a hard time understanding why I always get the error `startify: Can't create 'last used session' symlink.` while saving or loading a session using startify.

Saving and loading are both successful, but it's annoying having to confirm the error every time.

Maybe someone could think of what could be wrong.
Thanks!

**Comment by mhinz** [comment]
On non-Windows systems, Startify _should_ automatically create a symlink in your session directory (`~/.vim/session` by default), when a session is loaded or saved.

That way `:SL` without arguments can load the last used session.

But apparently creating the symlink fails on your system.

Could you please add the following line above [this line](https://github.com/mhinz/vim-startify/blob/master/autoload/startify.vim#L866-Lundefined):

```vim
redraw | echomsg 'DEBUG: '. cmd
```

... restart and try loading a session again. When it fails, you should see that message in your command-line. (If not, check `:messages` to see the message history.)

**Comment by kadfak** [comment]
The symlink gets created successfully and it points to the correct session file. 

Here's the output of what you gave:
`DEBUG: ln -sf 'test' '/home/x/.vim/session/__LAST__'`

When using `:SL` without args, the error does not appear. Loading by specifying the session or from the startify launcher and saving do print the error.

**Comment by mhinz** [comment]
Hmm, if the link creation succeeds and the message gets shown anywy, if can only mean that either your `ln` is return a number that isn't 0 or a Vim bug.

I'll remove the message, but out of curiosity.. what's is the output of:

```
$ ln -sf foo bar
$ echo $?
```

**Comment by kadfak** [comment]
The output of
```
$ ln -sf foo bar
$ echo $?
```
is 0

I also printed the output of `v:shell_error` after the `system()` call in `startify.vim` - there it gives -1.
If I removed `silent!` from `silent! call system(cmd)`, the output changed to 0.
I don't know why, but to me the problem seems to be `silent!`.

**Comment by mhinz** [comment]
OK, so apparently `:silent` can really influence `v:shell_error`. But since `ln -sf` and `system()` are usually quiet on their own already, we can remove the more or less superfluous `:silent`.

(I also reverted the commit that removed the warning message.)

Does it work for you now?

**Comment by kadfak** [comment]
Yes, that way it works.
Thanks for your cooperation!

**Comment by mhinz** [comment]
Great, thanks for reporting the issue!

**Comment by jamessan** [comment]
This should be fixed upstream by [v8.0.0071].

[v8.0.0071]: https://github.com/vim/vim/releases/tag/v8.0.0071
